tracing-subscriber/src/filter/env/mod.rs

```diff
@@ -223,11 +229,14 @@ impl<S: Subscriber> Layer<S> for EnvFilter {
 
     fn enabled(&self, metadata: &Metadata<'_>, _: Context<'_, S>) -> bool {
         let level = metadata.level();
+        let target = metadata.target();
         self.scope
             .with(|scope| {
                 for filter in scope.iter() {
-                    if filter >= level {
-                        return true;
+                    if &filter.level >= level {
+                        if filter.target.matches(&target) {
```

Comment:
It occurs to me that if we think the nested `if` is ugly or unclear, this could be simplified to
```rust
if &filter.level >= level && filter.target.matches(&target) {
    return true;
}
```

Happy to merge it regardless, though.

Reply:
Let me do this.